src/bson.c

```diff
@@ -470,14 +515,41 @@ static bool php_phongo_bson_visit_null(const bson_iter_t* iter ARG_UNUSED, const
 	return false;
 } /* }}} */
 
+/* qsort() compare callback for alphabetizing regex flags upon initialization */
+static int php_phongo_regex_compare_flags(const void* f1, const void* f2)
+{
+	if (*(const char*) f1 == *(const char*) f2) {
+		return 0;
+	}
+
+	return (*(const char*) f1 > *(const char*) f2) ? 1 : -1;
+}
+
+static void php_phongo_bson_new_regex_from_regex_and_options(zval* object, const char* pattern, const char* flags TSRMLS_DC) /* {{{ */
+{
+	php_phongo_regex_t* intern;
+
+	object_init_ex(object, php_phongo_regex_ce);
+
+	intern              = Z_REGEX_OBJ_P(object);
+	intern->pattern_len = strlen(pattern);
+	intern->pattern     = estrndup(pattern, intern->pattern_len);
+	intern->flags_len   = strlen(flags);
+	intern->flags       = estrndup(flags, intern->flags_len);
+
+	/* Ensure flags are alphabetized upon initialization. This may be removed
+	 * once CDRIVER-1883 is implemented. */
+	qsort((void*) intern->flags, intern->flags_len, 1, php_phongo_regex_compare_flags);
```

Comment:
[CDRIVER-1883](https://jira.mongodb.org/browse/CDRIVER-1883) was fixed in 1.7.0. Can this and `php_phongo_regex_compare_flags` be removed?

This code was originally added in https://github.com/mongodb/mongo-php-driver/commit/e9271bf8a9919e6e85cc9372708f0841aa4fcd00 for [PHPC-885](https://jira.mongodb.org/browse/PHPC-885). Looking at that issue, the BSON corpus tests should provide coverage for this behavior.